plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/configuration/DatabaseMetadata.java

```diff
@@ -38,20 +38,35 @@ public int getVersion() {
     return version;
   }
 
-  public static DatabaseMetadata fromDirectory(final Path databaseDir) throws IOException {
-    final File metadataFile = getDefaultMetadataFile(databaseDir);
+  public static DatabaseMetadata lookUpFrom(final Path databaseDir, final Path dataDir)
+      throws IOException {
+    File metadataFile = getDefaultMetadataFile(dataDir);
+    final boolean shouldLookupInDatabaseDir = !metadataFile.exists();
+    if (shouldLookupInDatabaseDir) {
+      metadataFile = getDefaultMetadataFile(databaseDir);
+    }
+    DatabaseMetadata databaseMetadata;
     try {
-      return MAPPER.readValue(metadataFile, DatabaseMetadata.class);
+      databaseMetadata = MAPPER.readValue(metadataFile, DatabaseMetadata.class);
     } catch (FileNotFoundException fnfe) {
-      return new DatabaseMetadata(0);
+      databaseMetadata = new DatabaseMetadata(0);
     } catch (JsonProcessingException jpe) {
       throw new IllegalStateException(
           String.format("Invalid metadata file %s", metadataFile.getAbsolutePath()), jpe);
     }
+    if (shouldLookupInDatabaseDir) {
+      writeToDirectory(databaseMetadata, dataDir);
```

Comment:
If we migrate the database metadata we should totally log this at INFO level, if not WARN.

Reply:
good idea